sdk/containerregistry/Azure.Containers.ContainerRegistry/src/Blob/ContainerRegistryBlobClient.cs

```diff
@@ -0,0 +1,503 @@
+                Response rawResponse = response.GetRawResponse();
+                ManifestMediaType mediaType = rawResponse.Headers.ContentType;
+
+                if (!rawResponse.Headers.TryGetValue("Docker-Content-Digest", out string registryDigest))
+                {
+                    throw _clientDiagnostics.CreateRequestFailedException(rawResponse, "Response did not contain \"Docker-Content-Digest\" header.");
+                }
+
+                if (!ValidateDigest(rawResponse.ContentStream, registryDigest))
+                {
+                    throw _clientDiagnostics.CreateRequestFailedException(rawResponse, "The manifest's digest according to the registry does not match the locally computed digest value.");
+                }
+
+                using var document = JsonDocument.Parse(rawResponse.ContentStream);
+                var manifest = OciManifest.DeserializeOciManifest(document.RootElement);
+                return Response.FromValue(new DownloadManifestResult(registryDigest, manifest), rawResponse);
+            }
+            catch (Exception e)
+            {
+                scope.Failed(e);
+                throw;
+            }
+        }
+
+        /// <summary>
+        /// Download the manifest for an OCI Artifact.
+        /// </summary>
+        /// <param name="digest">The digest of the manifest to download.</param>
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        /// <returns></returns>
+        public virtual async Task<Response<DownloadManifestResult>> DownloadManifestAsync(string digest, CancellationToken cancellationToken = default)
+        {
+            Argument.AssertNotNullOrEmpty(digest, "digest");
+
+            using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(ContainerRegistryBlobClient)}.{nameof(DownloadManifest)}");
+            scope.Start();
+            try
+            {
+                Response<ManifestWrapper> response = await _restClient.GetManifestAsync(_repositoryName, digest, ManifestMediaType.OciManifest.ToString(), cancellationToken).ConfigureAwait(false);
+                Response rawResponse = response.GetRawResponse();
+                ManifestMediaType mediaType = rawResponse.Headers.ContentType;
+
+                if (!rawResponse.Headers.TryGetValue("Docker-Content-Digest", out string registryDigest))
+                {
+                    throw _clientDiagnostics.CreateRequestFailedException(rawResponse, "Response did not contain \"Docker-Content-Digest\" header.");
+                }
+
+                if (!ValidateDigest(rawResponse.ContentStream, registryDigest))
+                {
+                    throw _clientDiagnostics.CreateRequestFailedException(rawResponse, "The manifest's digest according to the registry does not match the locally computed digest value.");
```

Comment:
😮 What do we do if this happens?

Reply:
Fail!

This would be an indication of tampering or some other error, so I would assume the use case would be to report the error to the end-user (this would be someone trying to download Bicep modules), and I imagine they would need to contact support to investigate further.